internal/install/application_configuration.go

```diff
@@ -34,11 +34,28 @@ type stack struct {
 	ImageRefOverrides map[string]ImageRefs `yaml:"image_ref_overrides"`
 }
 
+func checkImageRefOverride(envVar string, fallback string) string {
+	refOverride := os.Getenv(envVar)
+	if refOverride == "" {
+		return fallback
+	}
+
+	return refOverride
+}
+
 func (s stack) ImageRefOverridesForVersion(version string) ImageRefs {
 	refs, ok := s.ImageRefOverrides[version]
+
+	kibanaRefOverride := checkImageRefOverride("KIBANA_IMAGE_REF_OVERRIDE", DefaultKibanaRef)
```

Comment:
Is there a chance that you will also need other Refs? for Elasticsearch and Agent? Maybe it's valid to introduce them as well.

Reply:
I have them in my local branch 😅